libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewModel.kt

```diff
@@ -99,34 +46,36 @@ class PreviewImageViewModel : ViewModel() {
         updateUiState(currentUiState.copy(viewPagerItemsStates = newImageUiStates))
     }
 
-    fun onLoadIntoImageViewFailed(url: String, position: Int) {
-        /*val newState = when (val currentState = uiState.value) {
-            is ImageDataStartLoadingUiState -> {
-                val lowResImageUrl = currentState.imageData.lowResImageUrl
-                if (url == lowResImageUrl) {
-                    ImageInLowResLoadFailedUiState
-                } else {
-                    ImageInHighResLoadFailedUiState
-                }
-            }
-            else -> ImageInHighResLoadFailedUiState
-        }
-
-        updateUiState(newState)*/
+    fun onLoadIntoImageViewFailed(currentUrl: String, currentPosition: Int) {
+        val newImageUiStates = updateViewPagerItemsUiStates(
+            currentUrl = currentUrl,
+            currentPosition = currentPosition,
+            loadSuccess = false
+        )
+        val currentUiState = uiState.value as UiState
+        updateUiState(currentUiState.copy(viewPagerItemsStates = newImageUiStates))
     }
 
-    fun onLoadIntoFileSuccess(inputFilePath: String) {
+    /*fun onLoadIntoFileSuccess(inputFilePath: String) {
         updateLoadIntoFileState(ImageLoadToFileSuccessState(inputFilePath))
         _navigateToCropScreenWithFileInfo.value = Pair(inputFilePath, outputFileExtension)
     }
 
     fun onLoadIntoFileFailed() {
         // TODO: Do we need to display any error message to the user?
         updateLoadIntoFileState(ImageLoadToFileFailedState)
-    }
+    }*/
 
-    fun onLoadIntoImageViewRetry(loResImageUrl: String, hiResImageUrl: String) {
-//        updateUiState(createImageDataStartLoadingUiState(loResImageUrl, hiResImageUrl))
+    @VisibleForTesting
```

Comment:
Do we really need the @VisibleForTesting annotation? Can't we invoke the retry method the same way the view/ViewHolder invokes it?

Reply:
It is actually invoked the same way. I was initially trying to call it explicitly when I added the annotation. Removed here: a8cbcda

https://github.com/wordpress-mobile/WordPress-Android/blob/6e8ce4e9f42ef3f0da4042e12de7341564f65655/libs/image-editor/ImageEditor/src/test/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewModelTest.kt#L247

Thanks for spotting it :)